test/test_transforms.py

```diff
@@ -734,6 +735,109 @@ def test_rotate(self):
 
         assert np.all(np.array(result_a) == np.array(result_b))
 
+    def test_affine(self):
+        x = np.zeros((100, 100, 3), dtype=np.uint8)
```

Comment:
instead of having an all-zero matrix, could you instead have a matrix which has some pattern in it (maybe even a random matrix).
Currently this test doesn't check for correctness because any multiplicative transform on a constant all-zero image will result in all zeros.

Reply:
@fmassa I'm not sure if I understood you correctly, but actually, the input image is not completely all zero but zero with 3 white points = edges of the letter L with the last point in the center of the image
```
        x = np.zeros((100, 100, 3), dtype=np.uint8)
        pts = [(40, 40), (50, 40), (50, 50)]
        cnt = [50, 50]
        for pt in pts:
            x[pt] = [255, 255, 255]
``` 
and in the test there is a check if manually warped points for example [here](https://github.com/vfdev-5/vision/blob/5940685d8aeb3dff686246762e4b4d25b8ae2312/test/test_transforms.py#L755) can be found in non-null locations of the warped image (warped with `F.affine`). 
Could you explain me how to check the transformation differently? Thanks

Reply:
Oh, I have missed the `L` that is in the image. That is what I meant, sorry for missing it before.
But I'm afraid that 3 pixels are not enough for the transformation, because there can be aliasing which would make it a big blur. Maybe you could make the `L` a bit bigger, say 10 pixels wide?

Reply:
You are right, with aliasing sometimes points disappeared (used Image.NEAREST and then changed to Image.BILINEAR). Okay, I can modify it to make it bigger. In this case there will be much more points to transform manually and then check them in the warped image.
What do you think about transformation checking? Now it happens like this
```
xs, ys = # manually transformed points
result = F.affine(...)
r, c, ch = np.where(result)
assert all(y in r for y in ys)
assert all(x in c for x in xs)
```
which means that I verify that ONLY manually transformed points should be in non-null image locations and not inverse.

Reply:
Having a closer look to your testing code, you only check that the manually verified points are in the result, but not that all points in the result are in the manually-verified set (so the result could be much larger).
A possibility would be to also assert on the other direction
```python
assert all(y in r for y in ys)
assert all(rs in y for rs in r)
```
etc.
What do you think?

Reply:
👍